pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -214,31 +215,19 @@ private String getMultiStageQueryResponse(String query, String queryOptions, Htt
       // tenants can be completely disjoint. The leaf stages which access segments will be processed on the respective
       // server tenants for each table. The intermediate stages can be processed in either or all of the server tenants
       // belonging to the tables.
-      brokerTenant = getCommonBrokerTenant(tableConfigList);
-      if (brokerTenant == null) {
+      brokerTenants = getCommonBrokerTenants(tableConfigList);
+      if (brokerTenants.isEmpty()) {
         return QueryException.getException(QueryException.BROKER_REQUEST_SEND_ERROR, new Exception(
             String.format("Unable to dispatch multistage query with multiple tables : %s " + "on different tenant",
                 tableNames))).toString();
       }
     } else {
       // TODO fail these queries going forward. Added this logic to take care of tautologies like BETWEEN 0 and -1.
-      List<String> allBrokerList = new ArrayList<>(_pinotHelixResourceManager.getAllBrokerTenantNames());
-      brokerTenant = allBrokerList.get(RANDOM.nextInt(allBrokerList.size()));
+      brokerTenants = _pinotHelixResourceManager.getAllBrokerTenantNames();
```

Comment:
This doesn't seem correct. We should pick a random tenant

Reply:
Actually, directly picking a random broker might be even better

Reply:
what would be the best solution here? i don't see a get random broker instance API in helix resource manager. should i add one?